firmware/targets/f7/furi_hal/furi_hal_subghz.c

```diff
@@ -510,7 +510,9 @@ static void furi_hal_subghz_async_tx_refill(uint32_t* buffer, size_t samples) {
             furi_hal_subghz_async_tx.callback(furi_hal_subghz_async_tx.callback_context);
 
         if(level_duration_is_wait(ld)) {
-            return;
+            *buffer = API_HAL_SUBGHZ_ASYNC_TX_GUARD_TIME;
```

Comment:
Going to cause periodic crashes.

Reply:
Details in:

```
            // Inject guard time if level is incorrect
            bool level = level_duration_get_level(ld);
            if(is_odd == level) {
                *buffer = API_HAL_SUBGHZ_ASYNC_TX_GUARD_TIME;
                buffer++;
                samples--;
                if(!level) {
                    furi_hal_subghz_async_tx.duty_high += API_HAL_SUBGHZ_ASYNC_TX_GUARD_TIME;
                } else {
                    furi_hal_subghz_async_tx.duty_low += API_HAL_SUBGHZ_ASYNC_TX_GUARD_TIME;
                }
                // This code must be invoked only once: when encoder starts with low level.
                // Otherwise whole thing will crash.
                furi_check(samples > 0);
            }
```